src/DAL/migration/functions.sql

```diff
@@ -0,0 +1,61 @@
+-- FUNCTION: PolygonParts._mc_merge_polygons_aggregator(geometry, geometry)
+
+-- DROP FUNCTION IF EXISTS "PolygonParts"._mc_merge_polygons_aggregator(geometry, geometry);
+
+CREATE OR REPLACE FUNCTION "PolygonParts"._mc_merge_polygons_aggregator(
+	state geometry,
+	geom geometry)
+    RETURNS geometry
+    LANGUAGE 'plpgsql'
+    COST 100
+    VOLATILE PARALLEL UNSAFE
+AS $BODY$
+BEGIN
+    IF state IS NULL THEN
+        RETURN geom;
+    ELSE
+		RETURN st_collect(st_collect(res), geom) FROM (
+			SELECT ST_Difference(geom_dump, geom) as res
```

Comment:
I see that we currently aren't merging parts from the same `multipolygon` if they intersect.
Not only that, doesn't this make us potentially lose data?

Reply:
irrelevant